packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/Field.tsx

```diff
@@ -0,0 +1,279 @@
+        },
+        {
+          item: `${name}.${currentIndex}`,
+          position: getItemPos(newIndex),
+        }
+      )
+    );
+
+    moveFieldRow(name, currentIndex, newIndex);
+  };
+
+  const getItemPos = (index: number) => `${index + 1} of ${value.length}`;
+
+  const handleCancel = (index: number) => {
+    setLiveText(
+      formatMessage(
+        {
+          id: getTranslation('dnd.cancel-item'),
+          defaultMessage: '{item}, dropped. Re-order cancelled.',
+        },
+        {
+          item: `${name}.${index}`,
+        }
+      )
+    );
+  };
+
+  const handleGrabItem = (index: number) => {
+    setLiveText(
+      formatMessage(
+        {
+          id: getTranslation('dnd.grab-item'),
+          defaultMessage: `{item}, grabbed. Current position in list: {position}. Press up and down arrow to change position, Spacebar to drop, Escape to cancel.`,
+        },
+        {
+          item: `${name}.${index}`,
+          position: getItemPos(index),
+        }
+      )
+    );
+  };
+
+  const handleDropItem = (index: number) => {
+    setLiveText(
+      formatMessage(
+        {
+          id: getTranslation('dnd.drop-item'),
+          defaultMessage: `{item}, dropped. Final position in list: {position}.`,
+        },
+        {
+          item: `${name}.${index}`,
+          position: getItemPos(index),
+        }
+      )
+    );
+  };
+
+  const handleRemoveComponent = (name: string, currentIndex: number) => () => {
+    removeFieldRow(name, currentIndex);
+  };
+
+  const renderButtonLabel = () => {
+    if (addComponentIsOpen) {
+      return formatMessage({ id: 'app.utils.close-label', defaultMessage: 'Close' });
+    }
+
+    // if (hasError && dynamicZoneError.id?.includes('max')) {
+    //   return formatMessage({
+    //     id: 'components.Input.error.validation.max',
+    //     defaultMessage: 'The value is too high.',
+    //   });
+    // }
+
+    // if (hasError && dynamicZoneError.id?.includes('min')) {
+    //   return formatMessage(
+    //     {
+    //       id: getTranslation(`components.DynamicZone.missing-components`),
+    //       defaultMessage:
+    //         'There {number, plural, =0 {are # missing components} one {is # missing component} other {are # missing components}}',
+    //     },
+    //     { number: missingComponentNumber }
+    //   );
+    // }
+
+    return formatMessage(
+      {
+        id: getTranslation('components.DynamicZone.add-component'),
+        defaultMessage: 'Add a component to {componentName}',
+      },
+      { componentName: label || name }
+    );
+  };
+
+  const ariaDescriptionId = React.useId();
+
+  return (
+    <DynamicZoneProvider isInDynamicZone>
+      <Flex direction="column" alignItems="stretch" gap={6}>
+        {dynamicDisplayedComponentsLength > 0 && (
+          <Box>
+            <DynamicZoneLabel
+              hint={hint}
+              label={label}
+              // labelAction={labelAction}
```

Comment:
Why do we drop the label action? should we have a TODO to put it back?

Reply:
I think it's fine, this is going into a feature branch so we can re-evaluate any comments, but no one else will be working on this branch on the FE at least.